src/main/java/com/scalar/db/storage/cosmos/MutateStatementHandler.java

```diff
@@ -42,42 +50,71 @@ public MutateStatementHandler(CosmosClient client, TableMetadataManager metadata
       List<Record> results = execute(operation);
 
       return results;
-    } catch (NoMutationException e) {
-      throw e;
-    } catch (RuntimeException e) {
-      LOGGER.error(e.getMessage());
-      throw new ExecutionException(e.getMessage(), e);
+    } catch (CosmosException e) {
+      throwException(e);
     }
+
+    return Collections.emptyList();
+  }
+
+  protected void executeStoredProcedure(Mutation mutation) throws CosmosException {
+    executeStoredProcedure(Arrays.asList(mutation));
   }
 
-  protected void executeStoredProcedure(String storedProcedureName, Mutation mutation)
-      throws CosmosException, NoMutationException {
-    Optional<Record> record = makeRecord(mutation);
-    String query = makeConditionalQuery(mutation);
-    List<Object> args =
-        record.isPresent() ? Arrays.asList(record.get(), query) : Arrays.asList(query);
+  protected void executeStoredProcedure(List<? extends Mutation> mutations) throws CosmosException {
+    List<Integer> types = new ArrayList<>();
+    List<Record> records = new ArrayList<>();
+    List<String> queries = new ArrayList<>();
+
+    mutations.forEach(
+        mutation -> {
+          types.add(getMutationType(mutation).ordinal());
+          records.add(makeRecord(mutation));
+          queries.add(makeConditionalQuery(mutation));
+        });
+    List<Object> args = new ArrayList<>();
+    args.add(mutations.size());
+    args.addAll(types);
+    args.addAll(records);
+    args.addAll(queries);
 
     CosmosStoredProcedureRequestOptions options =
         new CosmosStoredProcedureRequestOptions()
-            .setPartitionKey(new PartitionKey(getConcatenatedPartitionKey(mutation)));
+            .setPartitionKey(new PartitionKey(getConcatenatedPartitionKey(mutations.get(0))));
 
     CosmosStoredProcedure storedProcedure =
-        getContainer(mutation).getScripts().getStoredProcedure(storedProcedureName);
+        getContainer(mutations.get(0)).getScripts().getStoredProcedure(MUTATION_STORED_PROCEDURE);
     CosmosStoredProcedureResponse response = storedProcedure.execute(args, options);
```

Comment:
Is it fine to ignore the response ?
If so, maybe it is better to do it below.
```suggestion
    storedProcedure.execute(args, options);
```

Reply:
The response isn't used for now. I've removed the response. Thanks!